repology/parsers/parsers/cygwin.py

```diff
@@ -49,9 +50,24 @@ def iter_parse(self, path: str, factory: PackageFactory, transformer: PackageTra
 
                     verpkg = pkg.clone()
 
-                    version = packagedata['versions'][maturity][-1]
-                    verpkg.set_version(version.rsplit('-', 1)[0])
-                    verpkg.set_rawversion(version)
+                    raw_version = packagedata['versions'][maturity][-1]
+                    (version, release) = raw_version.rsplit('-', 1)
+
+                    # If release is just '0', that means someone
+                    # forgot it counts from 1, but if it starts with
+                    # '0', the rest indicates the pre-release version
+                    # (as per Fedora/repodata.py)
+                    if release.startswith('0') and len(release) > 1:
+                        match = re.fullmatch('.*((?:alpha|beta|rc|(?<=[^a-z])[ab])(\.?[0-9]+)?)', release)
+                        if match:
+                            # known pre-release schema
+                            version += '-' + match.group(1)
```

Comment:
Just realized, you'd need `verpkg.set_flags(PackageFlags.devel)` here as this is a pre-release version.

Reply:
Yeah.  I think maybe that is also missing from repodata.py then, which I copied from...